pallets/liquid-staking/src/strategy.rs

```diff
@@ -0,0 +1,160 @@
+            return Default::default();
+        }
+        let mut distributions: Vec<(DerivativeIndex, Balance)> = vec![];
+        let amount = input.checked_div(&length).unwrap_or_default();
+        for (index, _) in unlocking_amount.iter() {
+            distributions.push((*index, amount));
+        }
+
+        distributions
+    }
+}
+
+pub struct QueueStrategy;
+impl<Balance: BalanceT + FixedPointOperand> DistributionStrategy<Balance> for QueueStrategy {
+    fn get_bond_distributions(
+        active_bonded_amount: &mut Vec<(DerivativeIndex, Balance)>,
+        input: Balance,
+        cap: Balance,
+        min_bond_amount: Balance,
+    ) -> Vec<(DerivativeIndex, Balance)> {
+        //ascending sequence
+        active_bonded_amount.sort_by(|a, b| a.1.cmp(&b.1));
+
+        let mut distributions: Vec<(DerivativeIndex, Balance)> = vec![];
+        let mut remain = input;
+
+        for (index, bonded) in active_bonded_amount.iter() {
+            if remain.is_zero() {
+                break;
+            }
+            let amount = cap.saturating_sub(*bonded).min(remain);
+            if amount.is_zero() {
+                // `active_bonded_amount` is an ascending sequence
+                // if occurs an item that exceed the cap, the items after this one must all be exceeded
+                break;
+            }
+
+            if amount.saturating_add(*bonded) < min_bond_amount {
+                continue;
+            }
+
+            distributions.push((*index, amount));
+            remain = remain.saturating_sub(amount);
+        }
+
+        distributions
+    }
+
+    fn get_unbond_distributions(
+        active_bonded_amount: &mut Vec<(DerivativeIndex, Balance)>,
+        input: Balance,
+        _cap: Balance,
+        min_bond_amount: Balance,
+    ) -> Vec<(DerivativeIndex, Balance)> {
+        // descending sequence
+        active_bonded_amount.sort_by(|a, b| b.1.cmp(&a.1));
+
+        let mut distributions: Vec<(DerivativeIndex, Balance)> = vec![];
+        let mut remain = input;
+
+        for (index, bonded) in active_bonded_amount.iter() {
+            if remain.is_zero() {
+                break;
+            }
+            let amount = remain.min(bonded.saturating_sub(min_bond_amount));
+            if amount.is_zero() {
+                continue;
+            }
+            distributions.push((*index, amount));
+            remain = remain.saturating_sub(amount);
+        }
+
+        distributions
+    }
+    fn get_rebond_distributions(
+        unlocking_amount: &mut Vec<(DerivativeIndex, Balance)>,
+        input: Balance,
+        _cap: Balance,
+        _min_bond_amount: Balance,
+    ) -> Vec<(DerivativeIndex, Balance)> {
+        // descending sequence
+        unlocking_amount.sort_by(|a, b| b.1.cmp(&a.1));
+
+        let mut distributions: Vec<(DerivativeIndex, Balance)> = vec![];
+        let mut remain = input;
+
+        for (index, unlocking) in unlocking_amount.iter() {
```

Comment:
maybe you can use `into_iter` to avoid using `*` in following code

Reply:
if we want to avoid using `*`, maybe we need to change the argument from `unbonding_amounts: &mut Vec<(DerivativeIndex, Balance, Balance)>,` to `unbonding_amounts: Vec<(DerivativeIndex, Balance, Balance)>,`